pkg/codegen/go/gen.go

```diff
@@ -4647,7 +4647,7 @@ func packageName(pkg *schema.Package) string {
 		info = goInfo
 	}
 	if info.RootPackageName != "" {
-		return info.RootPackageName
+		return strings.ReplaceAll(info.RootPackageName, "-", "")
```

Comment:
E.g. I know we have things like `makeValidIdentifier`-esque functions in various languages/places; not sure if we could have `MakeValidGoPackageName` etc. and use that here and elsewhere.

Reply:
Yeah, this should be using `goPackage()`, will fix.